src/box/memtx_space.c

```diff
@@ -1037,6 +1097,29 @@ memtx_build_on_replace(struct trigger *trigger, void *event)
 		if (stmt->old_tuple != NULL)
 			tuple_unref(stmt->old_tuple);
 	}
+	/*
+	 * Set on_rollback trigger on stmt to avoid
+	 * problem when rollbacked changes appears in
+	 * built-in-background index.
+	 */
+	struct on_rollback_trigger_with_data *on_rollback_associates =
+		region_aligned_alloc(&in_txn()->region,
```

Comment:
BTW there's `region_alloc_object`. But I'm not sure it'll simplify something here.